spec.md

```diff
@@ -523,13 +518,295 @@ schema being used.
+for the previously fetched sub-schemas and can just use the root schema. The
+Broker MUST provide the JSON Schema Object for the referenced URI in a subschema
+returned.
+
+Component Paths that use the scheme `osb_v2:` SHOULD be versioned, and the
+document behind the URI MUST NOT change after the catalog has been served to a
+Platform. 
+
+Platforms are able to leverage this JSON Schema by using or creating a library
+that supports pre-fetched JSON Schemas.
+
+Brokers MUST NOT allow duplicate JSON Pointers to be formed in any JSON Schema.
+
+A note about the scheme, at the time of this writing, there were no found client
+libraries that fully support the [JSON Schema](http://json-schema.org/) spec in
+the usage of URI for `$ref`. Only URL was supported in libraries. So as a
+compromise the `osb_v2:` scheme SHOULD be followed by three forward slashes
+(`///`). This indicates the authority is implied (authority is ''). 
+
+
+#### $ref Usage
+
+Brokers MAY leverage `$ref` and external JSON Schemas only if the JSON Schema
+references can be found within the [JSON Schemas](#json-schemas) API endpoint. $ref
+usage is assumed to be delegation (linking), not inclusion (replacement).
+Libraries that implement `$ref` content replacement result in adverse behaviors.
+
+For Brokers that support `$ref`:
+
+  * MUST respond with all JSON Schema documents for `/schemas` requests that are
+    referenced in the catalog.
+  * MAY provide `$ref`'s with the `osb_v2:` scheme, then
+    * MUST respond to `/schemas/:component_id` requests.
+
+### Request
+
+#### Route
+`GET /v2/catalog/schemas`
+
+#### Headers
+
+The following HTTP Headers are defined for this operation:
+
+| Header | Type | Description |
+| --- | --- | --- |
+| X-Broker-API-Version* | string | See [API Version Header](#api-version-header). |
+
+\* Headers with an asterisk are REQUIRED.
+
+#### cURL
+
+```
+$ curl http://username:password@service-broker-url/v2/catalog/schemas -H "X-Broker-API-Version: 2.13"
+```
+
+### Response
+
+| Status Code | Description |
+| --- | --- |
+| 200 OK | MUST be returned upon successful processing of this request. The expected response body is below. |
+
+
+#### Body
+
+The root JSON Schema document is a complete collection of JSON Schemas
+leveraged by the Broker's catalog. The Platform will process this
+document before catalog `$ref` URIs map to definitions. The body MUST contain
+all JSON Schema object definitions found in the catalog and referenced
+sub-schemas.
+
+| Response field | Type | Description |
+| --- | --- | --- |
+| $schema* | string | The JSON Schema declaring the version of JSON schema being used. |
+| $id* | string | The top level JSON Schema ID for this document. |
+| definitions* | Dictionary-of-JSON Schema Object | A collection of JSON sub-Schema Objects. |
+
+Please refer to the [JSON Schema](http://json-schema.org/) specification for
+more details on JSON Schema.
+
+Note the same restriction on `$ref` applies to `/schemas`.
```

Comment:
Which restrictions? Perhaps mention `/schema` where ever those restrictions are, or repeat them here.

Reply:
I was trying to say go read the `$ref Usage` section, but it looks like that section applies here too so I am goign to just leave this line out of the spec.